Expensify/App
# Onyx File handling - preparation

cc @thienlnam 

### Details

This is a proof of concept work that showcases file handling in Onyx as per the linked issue
Onyx was updated with peer dependency to [localforage](https://localforage.github.io/localForage/)
Onyx PR: https://github.com/Expensify/react-native-onyx/pull/102

Deleted `listenToStorageEvents` lib - this is now handled using IndexedDb and `localforage` in Onyx
Deleted `SignOutManager` lib - the updated `Onyx.clear` logic is synchronized across tabs and makes it unnecessary

### Fixed Issues
$ https://github.com/Expensify/App/issues/3867

### Tests

~~#### For all platforms~~

~~1. Go offline
2. Open a conversation
3. Select and send an attachment
4. Terminated the app
5. Resume connectivity
6. Open the app again
7. Go to the conversation and see the file appearing after it's uploaded
    - depending on how big is the file or how fast you do the last step - the file might already be uploaded~~

Edit: _The change needed for this was reverted 
It will be added and persisting attachments would work after we resolve: https://github.com/Expensify/App/pull/6556_

#### Regression check for Web

##### Multiple Tabs 
1. Open multiple tabs 
2. Open the same conversation on multiple tabs
3. Write a draft message (without sending it)
4. Observe the message is update across different tabs

##### Multiple Tabs (Signing out)
1. Again on multiple tabs
2. Log out from one tab
3. You should be logged out on all tabs
4. You should be able to log back in without problems (from the tab you logged out from or from another tab)

### QA Steps


### Tested On


### Screenshots


#### Web

##### No regressions on persisting plain messages 
https://user-images.githubusercontent.com/12156624/146098340-8d4cd484-d3a3-4144-a2a0-02bb7756d177.mp4

##### Using multiple tabs/windows 
https://user-images.githubusercontent.com/12156624/140195176-0a584ac7-648c-45e1-8dc5-3b0f7a4ef15c.mp4

#### Mobile Web

https://user-images.githubusercontent.com/12156624/146100054-481fbcc3-6874-4bcd-88b3-050fa7a8353b.mov

#### Desktop


https://user-images.githubusercontent.com/12156624/146099685-b0c918bc-2fc2-4e70-9b9b-fae701d509eb.mov

#### iOS

https://user-images.githubusercontent.com/12156624/146099992-b5a2b9a7-1a0a-4765-af20-1baa506b554d.mov

#### Android

https://user-images.githubusercontent.com/12156624/146101129-6b45dad2-9ec4-4eb3-9cd9-00cc1a44a2e8.mp4
